sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filtering/FilteringConfiguration.java

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.filtering;
+
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.swagger.models.DerivedMetricInfo;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.swagger.models.FilterConjunctionGroupInfo;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.swagger.models.CollectionConfigurationInfo;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.swagger.models.DocumentStreamInfo;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.swagger.models.DocumentFilterConjunctionGroupInfo;
+
+import java.util.Set;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashSet;
+import java.util.HashMap;
+
+public class FilteringConfiguration {
+    private final Set<String> seenMetricIds = new HashSet<>();
+
+    // key is the telemetry type
+    private final Map<String, List<DerivedMetricInfo>> validDerivedMetricInfos;
+
+    // first key is the telemetry type, second key is the id associated with the document filters
+    private final Map<String, Map<String, List<FilterConjunctionGroupInfo>>> validDocumentFilterConjunctionGroupInfos;
+
+    private final String etag;
+
+    public FilteringConfiguration() {
+        validDerivedMetricInfos = new HashMap<>();
+        validDocumentFilterConjunctionGroupInfos = new HashMap<>();
+        etag = "";
+    }
+
+    public FilteringConfiguration(CollectionConfigurationInfo configuration) {
+        validDerivedMetricInfos = new HashMap<>();
+        validDocumentFilterConjunctionGroupInfos = new HashMap<>();
+        etag = configuration.getETag();
+        parseDocumentFilterConfiguration(configuration);
+        parseMetricFilterConfiguration(configuration);
```

Comment:
to get visibility (to other threads) guarantee provided by immutable Java objects, you can't update the inner contents of fields after you initialize them, so in this case you'd want to do something like this:

```suggestion
        validDerivedMetricInfos = parseMetricFilterConfiguration(configuration);
        validDocumentFilterConjunctionGroupInfos = parseDocumentFilterConfiguration(configuration);
        etag = configuration.getETag();
```